### PR TITLE
table partition: add more warnings to inform the user that setting TiDB partition prune mode to dynamic at session level

### LINKED ISSUE #35042: Add more warnings to inform the user that setting TiDB partition prune mode to dynamic at session level
## Enhancement

if we set tidb_partition_prune_mode='dynamic', it will show a warning "Please analyze all partition tables again for consistency between partition and global stats", this warning kinds of implying that "if you analyze all partition table again, everything will work fine"? 

But the fact is, if the partition mode is set at session level, even after analyzing all partition tables again, TiDB will meet some other problems, so maybe we should add more warnings to inform the user that setting TiDB partition prune mode to dynamic at session level is not recommended in production environment . 

If the user wants to use dynamic partition prune in a production environment, they should set partition prune mode to dynamic at global level .

### PR DESCRIPTION
**Pull request by ymkzpx**
### What problem does this PR solve?

Issue Number: close #35042

Problem Summary:

### Check List

Tests 

- [x] Unit test

Side effects

None.

Documentation

None.

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None.
```

### CODE REVIEW

File: table/tables/partition_test.go [hunk 1]
```diff
@@ -729,3 +729,14 @@ func TestIssue31721(t *testing.T) {
 	tk.MustExec("insert into t_31721 values ('1')")
 	tk.MustExec("select * from t_31721 partition(p0, p1) where col1 != 2;")
 }
+
+func TestPruneMode(t *testing.T) {
```
Comment (bb7133) [inline table/tables/partition_test.go:733]:
```suggestion
func TestPruneModeWarningInfo(t *testing.T) {
```

File: table/tables/partition_test.go [hunk 2]
```diff
@@ -729,3 +729,14 @@ func TestIssue31721(t *testing.T) {
...
+	store, _, clean := testkit.CreateMockStoreAndDomain(t)
+	defer clean()
+
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("set @@tidb_partition_prune_mode = 'static'")
```
Comment (bb7133) [inline table/tables/partition_test.go:738]: Please add a check `tk.MustQuery("show warnings")` here to verify that no warning is reported.

File: table/tables/partition_test.go [hunk 3]
```diff
@@ -729,3 +729,14 @@ func TestIssue31721(t *testing.T) {
...
+	tk.MustExec("set session tidb_partition_prune_mode = 'dynamic'")
+	tk.MustQuery("show warnings").Sort().Check(testkit.Rows("Warning 1105 Please analyze all partition tables again for consistency between partition and global stats",
+		"Warning 1105 Please avoid setting partition prune mode to dynamic at session level and set partition prune mode to dynamic at global level"))
```
Comment (bb7133) [inline table/tables/partition_test.go:741]: Please add the test case for `set global tidb_partition_prune_mode`.

**Review by tiancaiamao — APPROVED** [review]

**Review by bb7133 — APPROVED** [review]
LGTM

File: sessionctx/variable/sysvar.go [hunk 4]
```diff
@@ -1406,6 +1406,7 @@ var defaultSysVars = []*SysVar{
 		newMode := strings.ToLower(strings.TrimSpace(val))
 		if PartitionPruneMode(s.PartitionPruneMode.Load()) == Static && PartitionPruneMode(newMode) == Dynamic {
 			s.StmtCtx.AppendWarning(errors.New("Please analyze all partition tables again for consistency between partition and global stats"))
+			s.StmtCtx.AppendWarning(errors.New("Please avoid setting partition prune mode to dynamic at session level and set partition prune mode to dynamic at global level"))
```
Comment (mjonss) [inline sessionctx/variable/sysvar.go:1409]: It is still not clear why one should use global level in the warning. Maybe a link to documentation to explain the reason?

**Review by mjonss — APPROVED** [review]

**Pull request merged**